File: app/src/main/java/com/juawapps/newsspread/di/AppModuleApplication.java
package com.juawapps.newsspread.di;

import android.content.Context;

import com.juawapps.newsspread.NewSpreadApplication;
import com.juawapps.newsspread.data.api.ApiProvider;
import com.juawapps.newsspread.data.api.NewsapiService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Database module
 */
@Module
public class AppModuleApplication {

    @Provides
    @Singleton
    Context provideContext(NewSpreadApplication application) {
        return application;
    }
}

File: app/src/main/java/com/juawapps/newsspread/di/AppModuleApplication.kt
package com.juawapps.newsspread.di

import android.content.Context

import com.juawapps.newsspread.NewSpreadApplication

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Database module
 */
@Module
class AppModuleApplication {

    @Provides
    @Singleton
    internal fun provideContext(application: NewSpreadApplication): Context {
        return application
    }
}
